aten/src/ATen/native/Loss.cpp

```diff
@@ -295,38 +295,45 @@ Tensor soft_margin_loss(
   return output;
 }
 
-Tensor smooth_l1_loss(const Tensor& input, const Tensor& target, const int64_t reduction) {
+Tensor smooth_l1_loss(const Tensor& input, const Tensor& target, const int64_t reduction, double beta) {
   Tensor loss;
   auto iter = TensorIterator::binary_op(loss, input, target);
-  smooth_l1_stub(iter.device_type(), iter);
+  smooth_l1_stub(iter.device_type(), iter, beta);
   return apply_loss_reduction(iter.output(), reduction);
 }
 
-Tensor& smooth_l1_loss_out(Tensor& result, const Tensor& input, const Tensor& target, int64_t reduction) {
+Tensor& smooth_l1_loss_out(Tensor& result, const Tensor& input, const Tensor& target, int64_t reduction, double beta) {
   if (reduction != Reduction::None) {
-    result = at::smooth_l1_loss(input, target, reduction);
+    Tensor loss;
+    auto iter = TensorIterator::binary_op(loss, input, target);
```

Comment:
@gchanan I updated smooth_l1_loss_out to something similar to what MSE does:
- make a TensorIterator that puts the smooth l1 loss output in a temporary tensor
- use the existing `<reduction>_out` functions to put the output of the reduction in the out var.

My educated guess as to why we need the temporary variable is because `result` has a different tensor shape than the tensor output of the actually kernel function when the caller is using reductions.